cmd/ctl/pkg/renew/renew.go

```diff
@@ -0,0 +1,249 @@
+func (o *Options) Run(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
+	restConfig, err := f.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+
+	cmClient, err := cmclient.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+
+	nss := []corev1.Namespace{{ObjectMeta: metav1.ObjectMeta{Name: o.Namespace}}}
+
+	// TODO: handle network context
+
+	if o.AllNamespaces {
+		kubeClient, err := kubernetes.NewForConfig(restConfig)
+		if err != nil {
+			return err
+		}
+
+		nsList, err := kubeClient.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+
+		nss = nsList.Items
+	}
+
+	var crts []cmapi.Certificate
+	for _, ns := range nss {
+		switch {
+		case o.All, len(o.LabelSelector) > 0:
+			crtsList, err := cmClient.CertmanagerV1alpha2().Certificates(ns.Name).List(context.TODO(), metav1.ListOptions{
+				LabelSelector: o.LabelSelector,
+			})
+			if err != nil {
+				return err
+			}
+
+			crts = append(crts, crtsList.Items...)
+
+		default:
+			for _, crtName := range args {
+				crt, err := cmClient.CertmanagerV1alpha2().Certificates(ns.Name).Get(context.TODO(), crtName, metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+
+				crts = append(crts, *crt)
+			}
+		}
+	}
+
+	if len(crts) == 0 {
+		if o.AllNamespaces {
+			fmt.Fprintln(o.ErrOut, "No Certificates found")
+		} else {
+			fmt.Fprintf(o.ErrOut, "No Certificates found in %s namespace.\n", o.Namespace)
+		}
+
+		return nil
+	}
+
+	for _, crt := range crts {
+		if err := o.renewCertificate(cmClient, &crt); err != nil {
+			return err
+		}
+	}
+
+	if o.Wait {
+		if err := o.waitCertificatesReady(cmClient, crts); err != nil {
+			return err
+		}
+
+		fmt.Fprintf(o.Out, "%d Certificates successfully renewed\n", len(crts))
+	}
+
+	return nil
+}
+
+func (o *Options) renewCertificate(cmClient *cmclient.Clientset, crt *cmapi.Certificate) error {
+	apiutil.SetCertificateCondition(crt, cmapi.CertificateConditionIssuing, cmmeta.ConditionTrue, "ManuallyTriggered", "Certificate re-issuance manually triggered")
+	_, err := cmClient.CertmanagerV1alpha2().Certificates(crt.Namespace).UpdateStatus(context.TODO(), crt, metav1.UpdateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to trigger issuance of Certificate %s/%s: %v", crt.Namespace, crt.Name, err)
+	}
+	fmt.Fprintf(o.Out, "Manually triggered issuance of Certificate %s/%s\n", crt.Namespace, crt.Name)
+	return nil
+}
+
+func (o *Options) waitCertificatesReady(cmClient *cmclient.Clientset, crts []cmapi.Certificate) error {
+	// TODO: start poll time after all get requests?
+	ticker := time.NewTicker(o.PollTime)
+	defer ticker.Stop()
+
+	ctx := context.TODO()
+
+	if o.Timeout > 0 {
+		var cancel func()
+		ctx, cancel = context.WithTimeout(ctx, o.Timeout)
+		defer cancel()
+	}
+
+	for {
+		lenReady := 0
+
+		for _, crt := range crts {
+			crt, err := cmClient.CertmanagerV1alpha2().Certificates(crt.Namespace).Get(context.TODO(), crt.Name, metav1.GetOptions{})
```

Comment:
Instead of nesting all this within a `for` loop in-line, perhaps create a `waitForCertificateRequestCondition(stopCh <-chan struct{}, namespace, name string, condition cmapi.CertificateCondition` method which can be called with `go run`? We'll also need to collect 'results' (i.e. each channel being complete) too.